cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -810,12 +810,41 @@ protected Future<Void> podDisruptionBudgetV1Beta1() {
     }
 
     /**
-     * Prepares annotations for Kafka pods which are known only in the KafkaAssemblyOperator level. These are later
-     * passed to KafkaCluster where there are used when creating the Pod definitions.
+     * Prepares annotations for Kafka pods within a StrimziPodSet which are known only in the KafkaAssemblyOperator level.
+     * These are later passed to KafkaCluster where there are used when creating the Pod definitions.
      *
-     * @return  Map with Pod annotations
+     * @return Map with Pod annotations
      */
-    private Map<String, String> kafkaPodAnnotations(int brokerId, boolean storageAnnotation)    {
+    private Map<String, String> podSetPodAnnotations(int brokerId, Secret kafkaSecret) {
+        Map<String, String> podAnnotations = commonKafkaPodAnnotations(brokerId);
+
+        // Annotation of broker certificate hash
+        podAnnotations.put(KafkaCluster.ANNO_STRIMZI_BROKER_CERT_HASH, Util.hashStub(kafka.getBrokerCertificates(kafkaSecret, brokerId).toString()));
```

Comment:
It's a bit weird to use the `toString` to calculate the hash here. If I understand the problem correctly we only _really care_ when the certificate itself changes (true, right now the rest of the CertAndKey only changes when the cert changes, but let's not build in that assumption). I think it would be better to use the actual broker "thumbprint". As in:

* Get the certificate's ASN1 encoded form (which you can get from `X509Certificate#getEncoded()`)
* Hash it using SHA1.

This is the same thumbprint/fingerprint that things like openssl use so it's actually meaningful more broadly than what we're using here

Reply:
Done